### PR TITLE
dynamic token liquidity fix

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**

### CODE REVIEW

**Review by sasurobert — COMMENTED** [review]
looks ok.

can you add new tests ? especially for this case.

**Review by sasurobert — APPROVED** [review]

File: builtInFunctions/esdtNFTTransfer.go [hunk 1]
```diff
@@ -296,7 +296,12 @@ func (e *esdtNFTTransfer) processNFTTransferOnSenderShard(
 			return nil, err
 		}
 	} else {
-		err = e.esdtStorageHandler.AddToLiquiditySystemAcc(esdtTokenKey, esdtData.Type, nonce, big.NewInt(0).Neg(quantityToTransfer))
+		hasDynamicRole, err := hasDynamicRole(acntSnd, tickerID, e.marshaller)
```
Comment (iulianpascalau) [inline builtInFunctions/esdtNFTTransfer.go:299]: to improve the understanding of the code I would have named the `hasDynamicRole` to `keepMetadataOnZeroLiquidity`
Reason: we all know how the variable is populated (by calling the function hasDynamicRole) but is more obscure to understand what is the parameter of the AddToLiquiditySystemAcc function

File: builtInFunctions/multiESDTNFTTransfer.go [hunk 2]
```diff
@@ -432,7 +432,12 @@ func (e *esdtNFTMultiTransfer) transferOneTokenOnSenderShard(
 			return nil, err
 		}
 	} else {
-		err = e.esdtStorageHandler.AddToLiquiditySystemAcc(esdtTokenKey, esdtData.Type, transferData.ESDTTokenNonce, big.NewInt(0).Neg(transferData.ESDTValue))
+		hasDynamicRole, err := hasDynamicRole(acntSnd, esdtTokenKey, e.marshaller)
```
Comment (iulianpascalau) [inline builtInFunctions/multiESDTNFTTransfer.go:435]: to improve the understanding of the code I would have named the `hasDynamicRole` to `keepMetadataOnZeroLiquidity`
Reason: we all know how the variable is populated (by calling the function hasDynamicRole) but is more obscure to understand what is the parameter of the AddToLiquiditySystemAcc function

**Review by iulianpascalau — APPROVED** [review]

File: builtInFunctions/esdtNFTTransfer.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply builtInFunctions/esdtNFTTransfer.go:299]: Good idea, renamed.

File: builtInFunctions/multiESDTNFTTransfer.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply builtInFunctions/multiESDTNFTTransfer.go:435]: Good idea, renamed.

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**